base/fs_unix.h

```diff
@@ -57,25 +58,46 @@ size_t file_size(const std::string& path)
   return (stat(path.c_str(), &sts) == 0) ? sts.st_size: 0;
 }
 
-void move_file(const std::string& src, const std::string& dst)
+bool move_file(const std::string& src, const std::string& dst)
 {
-  int result = std::rename(src.c_str(), dst.c_str());
-  if (result != 0)
-    // TODO add errno into the exception
-    throw std::runtime_error("Error moving file");
+  return std::rename(src.c_str(), dst.c_str()) == 0;
 }
 
-void copy_file(const std::string& src, const std::string& dst, bool overwrite)
+// Don't secure for interrumptions, write in a temp?
+bool copy_file(const std::string& src, const std::string& dst, bool overwrite)
 {
-  throw std::runtime_error("Error copying file: unimplemented");
+  if(compare_filenames(src, dst) == 0) // Those are same?
+    return false;
+
+  if(is_file(dst) && !overwrite) // If dst exists, we need overwriting
+    return false;
+
+  auto src_file = open_file(src, "rb"), dst_file = open_file(dst, "wb");
```

Comment:
This can be in two lines (just to make it clear that we are opening two files):
```c++
auto src_file = open_file(src, "rb");
auto dst_file = open_file(dst, "wb");
```